frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance_view.dart

```diff
@@ -38,36 +50,18 @@ class ThemeSetting extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return Row(
-      children: [
-        Expanded(
-          child: FlowyText.medium(
-            LocaleKeys.settings_appearance_theme.tr(),
-            overflow: TextOverflow.ellipsis,
-          ),
-        ),
-        AppFlowyPopover(
-          direction: PopoverDirection.bottomWithRightAligned,
-          child: FlowyTextButton(
-            currentTheme,
-            fontColor: Theme.of(context).colorScheme.onBackground,
-            fillColor: Colors.transparent,
-            onPressed: () {},
-          ),
-          popupBuilder: (BuildContext context) {
-            return IntrinsicWidth(
-              child: Column(
-                mainAxisSize: MainAxisSize.min,
-                children: [
-                  _themeItemButton(context, BuiltInTheme.defaultTheme),
-                  _themeItemButton(context, BuiltInTheme.dandelion),
-                  _themeItemButton(context, BuiltInTheme.lavender),
-                ],
-              ),
-            );
-          },
+    return ThemeSettingDropDown(
+      label: LocaleKeys.settings_appearance_theme.tr(),
+      currentValue: currentTheme,
+      popupBuilder: (_) => IntrinsicHeight(
+        child: Column(
+          children: [
+            _themeItemButton(context, BuiltInTheme.defaultTheme),
+            _themeItemButton(context, BuiltInTheme.dandelion),
+            _themeItemButton(context, BuiltInTheme.lavender),
```

Comment:
same here. We can add MainAxisSize.min in Column and remove the IntrinsicHeight

Reply:
Haha yeah this is much better :)